pkg/storage/stores/local/metrics.go

```diff
@@ -0,0 +1,57 @@
+package local
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/weaveworks/common/instrument"
+)
+
+const (
+	statusFailure = "failure"
+	statusSuccess = "success"
+)
+
+type boltDBShipperMetrics struct {
+	// metrics for measuring performance of downloading of files per period initially i.e for the first time
+	initialFilesDownloadDurationSeconds *prometheus.GaugeVec
```

Comment:
I'm not sure if these are good metrics - won't they be too irregular? I expect it'll be hard to query these in prom if they're not regularly populated. I wonder if gauges which aren't updated regularly are still scraped regularly, hrm...

Also I'm wary of creating labels based on the table name.

Reply:
I agree that these metrics are irregular but they are there just to help understand slow queries which would most likely happen when downloading large index file. Without this metric we would have to rely on logs to find if query is slow because of downloading of files or some other contention is there.
Regarding labels based on table names, there would not be too many tables at a time. Also without that label, we would not have much visibility because otherwise the same metric would be updated for all the table downloads and the last one which was updated before prometheus scrapes would be visible, which would not be much of a use. What do you think?

Reply:
The other we have is adding a span in traces and getting rid of this metric.

Reply:
The other we have is adding a span in traces and getting rid of this metric.

Reply:
My concerns about sparsely populated gauges have been alleviated (this is handled via our instrumentation lib). 

It definitely seems a little weird to be creating per-table gauges that are only written on startup/first pull. This is even more costly if tables are/eventually become per-tenant. Could this instead be calculated across all tables? I find it hard to imagine us needing this level of granularity: we'll be pulling tables from the same object stores. I suspect having one gauge across all tables will be enough to help us and if we need to find _one_ problematic table, we can use logs. 

Did I explain that well?

Reply:
Yes, I understand your concern. The problem here is IndexClients get per table query so we can't have a metric across all tables downloaded for a query. Best option we have here is adding a span to the trace.

Reply:
I was thinking some shared struct which keeps individual download times/bytes for each table, then updates one gauge whenever a new table is downloaded. The gauge can be calculated from the sum of download times/bytes from each table.

Reply:
Something like 
```go
type DownloadTableMetrics struct{
  gauge *prometheus.Gauge
  tables map[string]struct{
    dur time.Duration
    bytes int
  }
}
```

Then when downloading a table you can do

`downloadTableMetrics.Add(tableName, downloadTime, byteCt)`, which will internally add that value then recalculate/update its underlying gauge.